### PR TITLE
security: Improved instructions on offline keys

### PR DESCRIPTION
**Pull request by StealthyCoder**
The information about the e-mail is dated, the e-mail notification no
longer contains the location to a one time download of TUF keys. This is
now possible through `fioctl keys rotate-root --initial` . The reference
to the first root key has been changed to `first` instead of `offline` .

Signed-off-by: Eric Bode <eric.bode@foundries.io>

### CODE REVIEW

**Comment by doanac** [comment]
Docs for 6eedcca867bf21460b17099d417bba9739412969 are browesable at: https://ci.foundries.io/projects/fio-docs/builds/1245/docs/artifacts/html/index.html

**Comment by doanac** [comment]
Docs for f83e578d033767b0635d8c08f670fb50a8e07889 are browesable at: https://ci.foundries.io/projects/fio-docs/builds/1246/docs/artifacts/html/index.html

File: source/reference-manual/security/authentication-xilinx.rst [hunk 1]
```diff
@@ -249,7 +249,7 @@ For more information on registering the PUF and how it is used by OP-TEE for gen
    https://www.xilinx.com/support/documentation/application_notes/xapp1333-external-storage-puf.pdf
 
 .. _bitstream-signed:
-   https://github.com/foundriesio/meta-lmp/blob/master/meta-lmp-bsp/dynamic-layers/xilinx-tools/recipes-bsp/bitstream/bitstream-signed_git.bb
+   https://github.com/foundriesio/meta-lmp/blob/master/meta-lmp-bsp/dynamic-layers/xilinx-tools/recipes-bsp/bitstream/bitstream-signed.bb
```
Comment (doanac) [inline source/reference-manual/security/authentication-xilinx.rst:253]: split that into its own commit.

**Comment by doanac** [comment]
@kprosise - its almost orthogonal to stated goal of this change, but I'm looking at this text:
```
To increase a root key's security further, it is encouraged that the Factory owner rotates it. Rotation will convert the root role's online-key, generated during the bootstrap of a Factory, to an offline key.
```

I'm not sure that's clear the "offline key" won't be known to Foundries and will only live on the customer's computer. I fear I'm too close to this problem to come up with a nice way to explain this. Do you have any thoughts/suggestions?

File: source/reference-manual/security/authentication-xilinx.rst (hunk 1 above)
Comment (StealthyCoder) [reply source/reference-manual/security/authentication-xilinx.rst:253]: I did that right? Or am I misunderstanding ? https://github.com/foundriesio/docs/pull/310/commits/f83e578d033767b0635d8c08f670fb50a8e07889 contains only this. :smile:

File: source/reference-manual/security/offline-keys.rst [hunk 2]
```diff
@@ -110,8 +98,8 @@ offline keys will look like::
     |-- config.json
     |-- credentials.zip
     |-- keys
-    |   |-- offline-root.pub     # Public root shown in root.json
-    |   |-- offline-root.sec     # The Factory's first root private
+    |   |-- first-root.pub     # Public root shown in root.json
+    |   |-- first-root.sec     # The Factory's first root private
```
Comment (doanac) [inline source/reference-manual/security/offline-keys.rst:102]: The initial content of this file has changed. I think this should just be:
```
tufrepo
    `-- keys
        |-- first-root.pub     # Public root shown in root.json
        |-- first-root.sec     # The Factory's first root private
        |-- fioctl-root-<keyid>.sec  # Your offline key(s)
        `-- fioctl-root-<keyid>.pub  
```

File: source/reference-manual/security/authentication-xilinx.rst (hunk 1 above)
Comment (vanmaegima) [reply source/reference-manual/security/authentication-xilinx.rst:253]: nothing major but there's a typo in this commit title

File: source/reference-manual/security/offline-keys.rst (hunk 2 above)
Comment (vanmaegima) [reply source/reference-manual/security/offline-keys.rst:102]: +1

Comment (vanmaegima) [reply source/reference-manual/security/offline-keys.rst:102]: +1

**Comment by doanac** [comment]
Docs for 2bc11074a51f8dd1f4fad17128dee4b267a5be5d are browesable at: https://ci.foundries.io/projects/fio-docs/builds/1247/docs/artifacts/html/index.html

File: source/reference-manual/security/offline-keys.rst (hunk 2 above)
Comment (StealthyCoder) [reply source/reference-manual/security/offline-keys.rst:102]: Changed the contents, hope I did it correctly. :+1:

**Comment by doanac** [comment]
Docs for 11155370812404f7d1e942e013346316a5620b3d are browesable at: https://ci.foundries.io/projects/fio-docs/builds/1248/docs/artifacts/html/index.html

File: source/reference-manual/security/offline-keys.rst (hunk 2 above)
Comment (doanac) [reply source/reference-manual/security/offline-keys.rst:102]: I think this is still off. i'm looking at my offline keys tarball and it only has:
```
$ tar -tzf tufkeys.tgz 
tufrepo/keys/first-root.pub
tufrepo/keys/fioctl-root-d14005323805e35381476e770c30d538d03a64e6d6aa652042121b345a492a22.pub
tufrepo/keys/fioctl-root-d14005323805e35381476e770c30d538d03a64e6d6aa652042121b345a492a22.sec
tufrepo/keys/first-root.sec
```
The other files referenced here like config.json have been removed.

Comment (StealthyCoder) [reply source/reference-manual/security/offline-keys.rst:100]: Updated tree view to only have keys under tufrepo.

Comment (doanac) [reply source/reference-manual/security/offline-keys.rst:100]: that's close, but its not rendering quite like the rest of the directory tree views in that page. I think you can copy/paste my suggestion above as-is and it should render properly.

**Comment by doanac** [comment]
Docs for a56ba9ccec95279bd4d66e64ea2d08a54b7c516e are browesable at: https://ci.foundries.io/projects/fio-docs/builds/1250/docs/artifacts/html/index.html

**Review by doanac — APPROVED** [review]
lets see what @kprosise thinks, but i like it

**Review by vanmaegima — APPROVED** [review]
LGTM

**Review by kprosise — APPROVED** [review]
LGTM

File: source/reference-manual/security/offline-keys.rst [hunk 3]
```diff
@@ -107,25 +95,13 @@ offline keys will look like::
...
-        |-- targets.json.checksum
-        `-- unsigned
-            `-- targets.json
-
-The critical file to keep from this tarball is ``offline-root.sec``.
+    `-- keys
+        |-- first-root.pub     # Public root shown in root.json
+        |-- first-root.sec     # The Factory's first root private
+        |-- fioctl-root-<keyid>.sec  # Your offline key(s)
+        `-- fioctl-root-<keyid>.pub
```
Comment (angolini) [inline source/reference-manual/security/offline-keys.rst:102]: I understand this is what goes inside the root.keys.tgz file. In this case, maybe it's better to replace "tufrepo"?

File: source/reference-manual/security/offline-keys.rst [hunk 4]
```diff
@@ -17,33 +17,21 @@ Every new Factory should have its keys rotated for offline storage.
...
+devices can trust. When a Factory is created, Foundries.io will create
+the first root key automatically.
+
+A new root key can be defined by doing a "key rotation".
+A key rotation sets the new root key for the factory,
 but it also signs the new root.json with the previous root key to
 demonstrate proper ownership. This can be done in fioctl with::
 
-    fioctl keys rotate-root /absolute/path/to/example.tgz
+   fioctl keys rotate-root --initial /absolute/path/to/root.keys.tgz
```
Comment (angolini) [inline source/reference-manual/security/offline-keys.rst:34]: Only from the text it's not clear to me if fiotcl downloads the root.keys.tgz to /absolute/path/to or if it creates the file, and them rotate the key from the factory to use the new one created.

Comment (angolini) [reply source/reference-manual/security/offline-keys.rst:34]: and I don't know if this detail is important. But I would like clarify if there is a prior command to download the root.keys.tgz or not

Comment (StealthyCoder) [reply source/reference-manual/security/offline-keys.rst:40]: Added the clarification detail

File: source/reference-manual/security/offline-keys.rst (hunk 3 above)
Comment (StealthyCoder) [reply source/reference-manual/security/offline-keys.rst:117]: If one would extract the tarball at that time there is a directory `tufrepo` containing a directory `keys` and so the `tufrepo` makes sense as the root of the tree in that way. I can also see it being confusing as the reference is not  to the file itself. I hopefully clarified it a bit in the leading text.

**Comment by doanac** [comment]
Docs for 4ee39500550beecf6b25ac18c338351283318cf4 are browesable at: https://ci.foundries.io/projects/fio-docs/builds/1253/docs/artifacts/html/index.html
